dtfs-central-api/api-tests/v1/deals/update-deal.api-test.js

```diff
@@ -0,0 +1,141 @@
+  updatedAt: 123456789,
+});
+
+describe('/v1/portal/deals', () => {
+  beforeAll(async () => {
+    await wipeDB.wipe([MONGO_DB_COLLECTIONS.DEALS, MONGO_DB_COLLECTIONS.FACILITIES]);
+  });
+
+  describe('PUT /v1/portal/deals/:id', () => {
+    it('returns the updated deal', async () => {
+      const postResult = await createDeal({ api, deal: newDeal, user: MOCK_PORTAL_USER });
+      const createdDeal = postResult.body;
+      const updatedDeal = {
+        ...newDeal,
+        _id: createdDeal._id,
+        additionalRefName: 'change this field',
+        eligibility: {
+          ...newDeal.eligibility,
+          mockNewField: true,
+        },
+      };
+
+      const expectedAuditRecord = generateParsedMockPortalUserAuditDatabaseRecord(MOCK_PORTAL_USER._id);
+      const expectedResponse = { ...updatedDeal, auditRecord: expectedAuditRecord };
+
+      const { status, body } = await api.put({ dealUpdate: updatedDeal, user: MOCK_PORTAL_USER }).to(`/v1/portal/deals/${createdDeal._id}`);
+
+      expect(status).toEqual(200);
+
+      expect(body).toEqual(expectAddedFieldsWithEditedBy({ baseDeal: expectedResponse, user: MOCK_PORTAL_USER, auditRecordType: 'portal' }));
+    });
+
+    it('handles partial updates', async () => {
+      const postResult = await createDeal({ api, deal: newDeal, user: MOCK_PORTAL_USER });
+      const createdDeal = postResult.body;
+
+      const partialUpdate = {
+        additionalRefName: 'change this field',
+        eligibility: {
+          mockNewField: true,
+        },
+      };
+
+      const expectedDataIncludingUpdate = {
+        ...newDeal,
+        _id: createdDeal._id,
+        additionalRefName: 'change this field',
+        eligibility: {
+          ...newDeal.eligibility,
+          mockNewField: true,
+        },
+      };
+
+      const { status: putStatus } = await api.put({ dealUpdate: partialUpdate, user: MOCK_PORTAL_USER }).to(`/v1/portal/deals/${createdDeal._id}`);
+      expect(putStatus).toEqual(200);
+
+      const { status, body } = await api.get(`/v1/portal/deals/${createdDeal._id}`);
+
+      expect(status).toEqual(200);
+      expect(body.deal).toEqual(expectAddedFieldsWithEditedBy({ baseDeal: expectedDataIncludingUpdate, user: MOCK_PORTAL_USER, auditRecordType: 'portal' }));
+    });
+
+    it('updates the deal', async () => {
+      const postResult = await createDeal({ api, deal: newDeal, user: MOCK_PORTAL_USER });
+      const createdDeal = postResult.body;
+
+      const updatedDeal = {
+        ...newDeal,
+        _id: createdDeal._id,
+        additionalRefName: 'change this field',
+      };
+
+      await api.put({ dealUpdate: updatedDeal, user: MOCK_PORTAL_USER }).to(`/v1/portal/deals/${createdDeal._id}`);
+
+      const { status, body } = await api.get(`/v1/portal/deals/${createdDeal._id}`);
+
+      expect(status).toEqual(200);
+
+      expect(body.deal).toEqual(expectAddedFieldsWithEditedBy({ baseDeal: updatedDeal, user: MOCK_PORTAL_USER, auditRecordType: 'portal' }));
```

Comment:
As commented previously, the fact `'portal'` isn't a constant nor in TS files makes me wary of maintainability